### PR TITLE
FFD code on JModelica

### LINKED ISSUE #612: FFD fails in JModelica tests due to unsupported OS
The test
`Buildings.ThermalZones.Detailed.Examples.FFD.NaturalConvectionWithControl` as well as all other FFD models fail in the JModelica tests with
```
FMIL: module = Model, log level = 3: [WARNING][FMU status:Warning]   <ModelicaError category="warning"><value name="msg">"Error: Unsupported operating system in cfdStartCosimulation.c.&#10;"</value></ModelicaError>
```

### PR DESCRIPTION
**Pull request by tsnouidui**
@mwetter This PR seems to address the FFD issues. 
Please review and merge to the master so we see what (JModelica) Jenkins says.
@tianwei1989 @zuowangda I refactored the C-code so it works with JModelica. I tested the code on Windows with JModelica 2.0.
The current code doesn't longer require to have `Buildings` as the current working directory.
You can see the changes in the pull request to understand how you need to set up models in the future.
This closes #612.

### CODE REVIEW

**Comment by mwetter** [comment]
@tsnouidui Please remove these files from the pull request that are not needed. I think there are 4 of them.

**Comment by tsnouidui** [comment]
@mwetter Two files managed to sneak in. I removed them. Sorry about that. Please go ahead and review the changes.

File: Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo [hunk 1]
```diff
@@ -87,9 +101,20 @@ Type <code>make all</code>
 </li>
 </ul>
 <p>
-please note to make use you have the right of executing the <code>compile.bat</code> when using method 2 on Windows.
-To check the availability right click the file and choose <code>Properties</code>. Go to <code>Security</code> tab and edit if you don't have the
+please note to make use you have the right of executing the <code>compile.bat</code>
```
Comment (mwetter) [inline Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo]: What does this sentence fragment mean? Please revise.

File: Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat [hunk 2]
```diff
@@ -0,0 +1,321 @@
+0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
```
Comment (mwetter) [inline Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat]: Why is this file needed? It looks like an output file?

**Comment by mwetter** [comment]
@tsnouidui Please see comments for incorrect documentation and
`Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat` seems to be an output file that is not needed. Please correct.

File: Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo (hunk 1 above)
Comment (tsnouidui) [reply Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo]: I am not sure what the original author meant with this and don't think it is even needed. However I can't reach them and wonder whether I should just delete the sentence or reformulate based on my best guess.

File: Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat (hunk 2 above)
Comment (tsnouidui) [reply Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat]: Those are what is called "obstacles files". They are needed to run the FFD code.

File: Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo (hunk 1 above)
Comment (mwetter) [reply Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo]: Then please reformulate it so users can understand it.

Comment (tsnouidui) [reply Buildings/ThermalZones/Detailed/Examples/FFD/UsersGuide.mo]: I decided to delete the sentence.

File: Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat (hunk 2 above)
Comment (mwetter) [reply Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat]: OK then, but a bit surprising that so many zeros are needed to simply run code. Is this Windows specific?

**Comment by tsnouidui** [comment]
@mwetter Please merge if you agree with my changes and comments.

**Comment by mwetter** [comment]
This is ready to merge when all tests pass.

File: Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat (hunk 2 above)
Comment (tsnouidui) [reply Buildings/Resources/Data/Rooms/FFD/NaturalConvectionWithControl.dat]: No. it is not Windows specific. There are bunch of such files in `Buildings/Resources/Data/Rooms/FFD. I could see them being retrieved in the C-code (`sci_reader.c`). The documentation says that they will be generated from a SCI program. the name of the original generated file is zeroone.dat so I guess they should have the zeros (maybe for the mesh definition?).

**Comment by zuowangda** [comment]
The 0 and 1 are indicators of obstacle or fluid for FFD mesh.
